server/src/services/checkSubmission.js

```diff
@@ -0,0 +1,174 @@
+    this.checkDuplicates(row8) &&
+    this.checkDuplicates(row9)
+    ){
+      return true
+    } else {
+      return false
+    }
+
+
+  }
+
+  checkColumn() {
+    let column1 = []
+    let column2 = []
+    let column3 = []
+    let column4 = []
+    let column5 = []
+    let column6 = []
+    let column7 = []
+    let column8 = []
+    let column9 = []
+
+    for(let x = 0; x< 7; x+3){
+      column1.push(this.sudokuPuzzle.box1[i])
+      column2.push(this.sudokuPuzzle.box1[i+1])
+      column3.push(this.sudokuPuzzle.box1[i+2])
+      column4.push(this.sudokuPuzzle.box2[i])
+      column5.push(this.sudokuPuzzle.box2[i+1])
+      column6.push(this.sudokuPuzzle.box2[i+2])
+      column7.push(this.sudokuPuzzle.box3[i])
+      column8.push(this.sudokuPuzzle.box3[i+1])
+      column9.push(this.sudokuPuzzle.box3[i+2])
+    }
+
+    for(let x= 0; x< 7; x+3){
+      column1.push(this.sudokuPuzzle.box4[i])
+      column2.push(this.sudokuPuzzle.box4[i+1])
+      column3.push(this.sudokuPuzzle.box4[i+2])
+      column4.push(this.sudokuPuzzle.box5[i])
+      column5.push(this.sudokuPuzzle.box5[i+1])
+      column6.push(this.sudokuPuzzle.box5[i+2])
+      column7.push(this.sudokuPuzzle.box6[i])
+      column8.push(this.sudokuPuzzle.box6[i+1])
+      column9.push(this.sudokuPuzzle.box6[i+2])
+    }
+
+    for(let x= 0; x< 7; x+3){
+      column1.push(this.sudokuPuzzle.box7[i])
+      column2.push(this.sudokuPuzzle.box7[i+1])
+      column3.push(this.sudokuPuzzle.box7[i+2])
+      column4.push(this.sudokuPuzzle.box8[i])
+      column5.push(this.sudokuPuzzle.box8[i+1])
+      column6.push(this.sudokuPuzzle.box8[i+2])
+      column7.push(this.sudokuPuzzle.box9[i])
+      column8.push(this.sudokuPuzzle.box9[i+1])
+      column9.push(this.sudokuPuzzle.box9[i+2])
+    }
+
+    if(!this.checkDuplicates(column1) &&
+    !this.checkDuplicates(column2) &&
+    !this.checkDuplicates(column3) &&
+    !this.checkDuplicates(column4) &&
+    !this.checkDuplicates(column5) &&
+    !this.checkDuplicates(column6) &&
+    !this.checkDuplicates(column7) &&
+    !this.checkDuplicates(column8) &&
+    !this.checkDuplicates(column9)
+    ){
+      return true
+    } else {
+      return false
+    }
+
+  }
+
+  checkBox() {
+
+    if(
+    !this.checkDuplicates(this.sudokuPuzzle.box1) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box2) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box3) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box4) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box5) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box6) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box7) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box8) &&
+    !this.checkDuplicates(this.sudokuPuzzle.box9)
+    )
+    {
+      return true
+    } else {
+      return false
+    }
+
+  }
+
+  get isCorrect() {
+    if(this.checkBox() && this.checkColumn() && this.checkRow() ){
+      return true
+    }
+    else{
+      return false
+    }
+  }
+
+}
```

Comment:
This was a LOT of work I am sure. Nice job.

Continue to brainstorm how this can be refactored though. There is a lot of duplication. 

I would have a method that returns all of the elements in the 1st, 2nd, 3rd, etc row of a puzzle, and the same for a column.